plugins/jobtype/src/azkaban/jobtype/HadoopSecureSparkWrapper.java

```diff
@@ -210,86 +301,33 @@ private static void handleDynamicResourceAllocation(String[] argArray) {
-    String executorMem = null;
-    String executorVcore = null;
-    String executorMemOverhead = null;
 
-    SparkConf sparkConf = new SparkConf(false);
-    sparkConf.setAll(Utils.getPropertiesFromFile(sparkPropertyFile));
+    SparkConf sparkConf = getSparkProperties();
 
     if (nodeLabelingYarn && nodeLabelingPolicy) {
-      for (int i = 0; i < argArray.length; i++) {
-        if (argArray[i] == null) {
-          continue;
-        }
-        if (nodeLabelingPolicy) {
-          // If yarn cluster enables node labeling, applications should be submitted to a default
-          // queue by a default conf(spark.yarn.queue) in spark-defaults.conf
-          // We should ignore user-specified queue param to enforece the node labeling
-          // (--queue test or --conf spark.yarn.queue=test)
-          if ((argArray[i].equals(SparkJobArg.SPARK_CONF_PREFIX.sparkParamName) &&
-               argArray[i+1].startsWith(SPARK_CONF_QUEUE))
-              || (argArray[i].equals(SparkJobArg.QUEUE.sparkParamName))) {
-
-            logger.info("Azbakan enforces node labeling. Ignore user param: "
-              + argArray[i] + " " + argArray[i+1]);
-            argArray[i] = null;
-            argArray[++i] = null;
-            continue;
-          }
-          if (autoNodeLabeling) {
-            // If auto node labeling is enabled, job type should ignore user supplied
-            // node label expression for Spark executors. This config will be automatically
-            // set by the job type based on the mem-to-vcore resource ratio requested by
-            // the user application.
-            if (argArray[i].equals(SparkJobArg.SPARK_CONF_PREFIX.sparkParamName) &&
-                argArray[i+1].startsWith(SPARK_EXECUTOR_NODE_LABEL_EXP)) {
-              logger.info("Azbakan auto-sets node label expression. Ignore user param: "
-                + argArray[i] + " " + argArray[i+1]);
-              argArray[i] = null;
-              argArray[++i] = null;
-              continue;
-            }
-            if (argArray[i].equals(SparkJobArg.EXECUTOR_CORES.sparkParamName)) {
-              executorVcore = argArray[++i];
-            }
-            if (argArray[i].equals(SparkJobArg.EXECUTOR_MEMORY.sparkParamName)) {
-              executorMem = argArray[++i];
-            }
-            if (argArray[i].equals(SparkJobArg.SPARK_CONF_PREFIX.sparkParamName) &&
-                argArray[i+1].startsWith(SPARK_EXECUTOR_MEMORY_OVERHEAD)) {
-              executorMemOverhead = argArray[i+1].split("=")[1].trim();
-            }
-          }
-        }
+      ignoreUserSpecifiedNodeLabelParameter(argArray, autoNodeLabeling);
+
+      // If yarn cluster enables node labelling, applications should be submitted to a default
+      // queue by a default conf(spark.yarn.queue) in spark-defaults.conf.
+      int queueParameterIndex = getUserSpecifiedQueueParameterIndex(argArray);
+      if (queueParameterIndex != -1) {
```

Comment:
Should remove this logic.
Enforcing node labels is the original queue enforcement mechanism.
With the dynamic-allocation aware queue enforcement, we should use the logic for handling queue enforcement, and handleNodeLabeling method should now only deals with auto node labeling.